jbd2: Fix oops with jbd-stats-through-procfs and external journal

From: Eric Sandeen <sandeen@redhat.com>

When using an external device for the journal,
jbd2_stats_proc_init() wants to use journal->j_dev in its call
to bdevname() but it's not assigned yet, resulting in an oops.

Signed-off-by: Eric Sandeen <sandeen@redhat.com>
Signed-off-by: Mingming Cao <cmm@us.ibm.com>
---

 fs/jbd2/journal.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)


diff --git a/fs/jbd2/journal.c b/fs/jbd2/journal.c
index 90b11ad..089d245 100644
--- a/fs/jbd2/journal.c
+++ b/fs/jbd2/journal.c
@@ -1034,11 +1034,11 @@ journal_t * jbd2_journal_init_dev(struct block_device *bdev,
 		journal = NULL;
 		goto out;
 	}
-	jbd2_stats_proc_init(journal);
 	journal->j_dev = bdev;
 	journal->j_fs_dev = fs_dev;
 	journal->j_blk_offset = start;
 	journal->j_maxlen = len;
+	jbd2_stats_proc_init(journal);
 
 	bh = __getblk(journal->j_dev, start, journal->j_blocksize);
 	J_ASSERT(bh != NULL);
